sunpy/sun/models.py

```diff
@@ -56,12 +57,14 @@
            10.157, 5.566, 2.259, 0.4483, 0.1528,
            0.042, 0.00361, 1.99e-7]
 
-_d = {'mass': _mass, 'luminosity': _luminosity, 'temperature': _temperature, 'density': _density}
-interior = pandas.DataFrame(_d, index = _radius)
-interior.index.name = 'radius'
-interior.units = {'radius': con.radius, 'mass': con.mass, 'luminosity': con.luminosity,
+_d = {'radius': _radius, 'mass': _mass, 'luminosity': _luminosity, 'temperature': _temperature, 'density': _density}
+_units = {'radius': con.radius, 'mass': con.mass, 'luminosity': con.luminosity,
                   'temperature': Quantity(1e6, 'K'), 'density': Quantity(1, 'g cm**-3')}
+interior = QTable(_d)
 interior.source = 'Turck-Chieze et al. (1988)'
+for labels in _units:
+    interior[labels].unit = _units[labels]
```

Comment:
I would suggest that instead of adding the units as a separate array, it would be better for the future if we add them in the data (so we can remove the comments that precedes them), so like:
```python
_density = [147.74, ...., 1.99e-7] * u.g/u.cm**3
```
So then, we don't need to loop around the units and it's easier to manipulate them.
This is how it's shown in [astropy's docs](http://docs.astropy.org/en/stable/table/mixin_columns.html#quantity-and-qtable)

Reply:
I understand. Will make that change.

Reply:
I don't think that will be practical. As trying to represent a unit like 10^6 K results in all values of the column being multiplied by 10^6 and the unit being K.
Example:

Reply:
![2019-02-27-13 49 13](https://user-images.githubusercontent.com/19383200/53475713-a0df3980-3a96-11e9-8bf1-9c6eb9afa4a3.jpg)